### ISSUE TITLE
All tabs lost in current session

### ISSUE THREAD

**Issue by Slion** [Slion/Fulguris#193]
In some rare situations current session can loose all tabs.
Consider writing session into a new file instead of overwriting the existing one.

This is happening a lot on Lineage OS 18.1 on F(x)tec Pro ¹.
In fact whenever the system restarts the app for you.
For instance change the System navigation settings when Fulguris is in the background.
Go back to it and current session is lost.

**Comment by Slion** [comment]
That's also happening when you have a session with around 150 tabs, send the app to the background and kill it asap through the recent task list. Basically that's killing the app while it's still writing the session file thus corrupting it and loosing all your tabs.

**Comment by Slion** [comment]
My assumption had been that the process is interrupted mid-save and you have a corrupt session file. So I took care of not saving directly to the session file but saving to another file then renaming it. To my surprise it did not help. Turns out there is a weird life cycle sequence that could trigger a shutdown before a saveState. So we were shooting ourselves in the foot. Deleting all tabs before saving the session.
